File: backend/app/services/gemini_service.py
from google import genai

from app.core.config import settings


class GeminiService:
    def __init__(self):
        self.client = genai.Client(api_key=settings.GEMINI_API_KEY)

    def generate_response(self, prompt: str) -> str:
        response = self.client.models.generate_content(
            model="gemini-2.0-flash",
            contents=prompt,
        )

        return response.text or "No response generated."

File: frontend/lib/features/ai/services/gemini_service.dart
import 'package:flutter_dotenv/flutter_dotenv.dart';
import 'package:google_generative_ai/google_generative_ai.dart';

class GeminiService {
  late final GenerativeModel _model;

  GeminiService() {
    final apiKey = dotenv.env['GEMINI_API_KEY'];

    if (apiKey == null || apiKey.isEmpty) {
      throw Exception("Gemini API key not found.");
    }

    _model = GenerativeModel(model: 'gemini-2.5-flash-lite', apiKey: apiKey);
  }

  Future<String> generateResponse(String prompt) async {
    try {
      final response = await _model.generateContent([Content.text(prompt)]);

      return response.text ?? "No response generated.";
    } catch (e) {
      throw Exception("Gemini Error: $e");
    }
  }
}
